app/models/affiliate.rb

```diff
@@ -0,0 +1,18 @@
+class Affiliate < ApplicationRecord
+
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+
+  has_many :coordinators
+  has_many :tutors
+
+  has_many :matches
+  has_many :students
+
+  validates :email, presence: true, length: { maximum: 255 },
+                    format: { with: VALID_EMAIL_REGEX },
+                    uniqueness: { case_sensitive: false }
+  validates :name, presence: true, length: { maximum: 100 }
+  validates :address, presence: true, length: { maximum: 200}
+  validates :phone_number, presence: true, length: { maximum: 11}
```

Comment:
the `1` is unnecessary. all phone numbers will be inside the united states. We can sanitize it on the way in. It should probably be either exactly 10 or exactly 11. I would vote for 10.